src/sql/workbench/contrib/views/browser/treeView.ts

```diff
@@ -770,7 +771,9 @@ abstract class AbstractTreeView extends Disposable implements ITreeView {
 				}
 			},
 
-			getActionsContext: () => (<TreeViewItemHandleArg>{ $treeViewId: this.id, $treeItemHandle: node.handle }),
+			getActionsContext: () => node.childProvider === mssqlProviderName // tracked change - need to pass treeItem for Data Explorer nodes
```

Comment:
Data Explorer is a general thing - not just MSSQL specific. Any extension can contribute a "data explorer" tree

https://github.com/microsoft/azuredatastudio/wiki/Servers-and-Data-Explorer-Tree-Views#data-explorer has some details about this

So this comment doesn't seem quite accurate with the change you're making here. 

A quick look around found this - which seems related. Although it's hard to tell at a quick glance what the correct behavior is here.

https://github.com/microsoft/azuredatastudio/blob/fed4f2a18efa25c437ec13effa5bddc045960b54/src/vs/workbench/api/common/extHostTreeViews.ts#L65

I'll try to take a look at this today to see if I can understand the root issue here.

Reply:
@cheenamalhotra Can you say exactly what you noticed was broken after your original fix?

Reply:
Just saw this now - the data explorer context menus don't work with the removal of treeItem, as payload from there was used in multiple command executions.